aws/resource_aws_cognito_user_group_test.go

```diff
@@ -0,0 +1,220 @@
+					resource.TestCheckResourceAttr("aws_cognito_user_group.main", "description", "This is the updated user group description"),
+					resource.TestCheckResourceAttr("aws_cognito_user_group.main", "precedence", "42"),
+					resource.TestCheckResourceAttrSet("aws_cognito_user_group.main", "role_arn"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSCognitoUserGroup_importBasic(t *testing.T) {
+	resourceName := "aws_cognito_user_group.main"
+	poolName := acctest.RandString(10)
+	groupName := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCognitoUserGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCognitoUserGroupConfig_basic(poolName, groupName),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSCognitoUserGroupExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		id := rs.Primary.ID
+		name := rs.Primary.Attributes["name"]
+		userPoolId := rs.Primary.Attributes["user_pool_id"]
+
+		if name == "" {
+			return errors.New("No Cognito User Group Name set")
+		}
+
+		if userPoolId == "" {
+			return errors.New("No Cognito User Pool Id set")
+		}
+
+		if id != fmt.Sprintf("%s/%s", userPoolId, name) {
+			return errors.New(fmt.Sprintf("ID should be user_pool_id/name. ID was %s. name was %s, user_pool_id was %s", id, name, userPoolId))
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).cognitoidpconn
+
+		params := &cognitoidentityprovider.GetGroupInput{
+			GroupName:  aws.String(rs.Primary.Attributes["name"]),
+			UserPoolId: aws.String(rs.Primary.Attributes["user_pool_id"]),
+		}
+
+		_, err := conn.GetGroup(params)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSCognitoUserGroupDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).cognitoidpconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_cognito_user_group" {
+			continue
+		}
+
+		params := &cognitoidentityprovider.GetGroupInput{
+			GroupName:  aws.String(rs.Primary.ID),
+			UserPoolId: aws.String(rs.Primary.Attributes["user_pool_id"]),
+		}
+
+		_, err := conn.GetGroup(params)
+
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+				return nil
+			}
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccAWSCognitoUserGroupConfig_basic(poolName, groupName string) string {
+	return fmt.Sprintf(`
+resource "aws_cognito_user_pool" "main" {
+  name = "%s"
+}
+
+resource "aws_cognito_user_group" "main" {
+  name         = "user-group-%s"
```

Comment:
Nitpick: It's usually better to keep the whole name (incl. the prefix like `user-group-` here) in the test body as we may then reference it in checks, instead of rebuilding it again. 😉